contracts/uniswap/KyberUniswapReserve.sol

```diff
@@ -312,4 +428,128 @@ contract KyberUniswapReserve is KyberReserveInterface, Withdrawable, Utils2 {
         kyberNetwork = _kyberNetwork;
         KyberNetworkSet(kyberNetwork);
     }
+
+    /*
+     * Uses amounts and rates to check if the reserve's internal inventory can
+     * be used directly.
+     *
+     * rateEthToToken and rateTokenToEth are in kyber rate format meaning
+     * rate as numerator and 1e18 as denominator.
+     */
+    function shouldUseInternalInventory(
+        ERC20 srcToken,
+        uint srcAmount,
+        ERC20 destToken,
+        uint destAmount,
+        uint rateSrcDest,
+        uint rateDestSrc
+    )
+        public
+        view
+        returns(bool)
+    {
+        require(srcAmount < MAX_QTY);
+        require(destAmount < MAX_QTY);
+
+        // Check for internal inventory balance limitations
+        ERC20 token;
+        uint rateEthToToken;
+        uint rateTokenToEth;
+        if (srcToken == ETH_TOKEN_ADDRESS) {
+            token = destToken;
+            uint tokenBalance = token.balanceOf(this);
+            if (
+                tokenBalance < destAmount ||
+                tokenBalance - destAmount < internalInventoryMin[token]
+            ) {
+                return false;
+            }
+            rateEthToToken = rateSrcDest;
+            // Normalize rate direction to enable comparison for checking spread
+            // and arbitrage: from ETH / Token -> Token / ETH
+            rateTokenToEth = 10 ** 36 / rateDestSrc;
+        } else {
+            token = srcToken;
+            if (this.balance < destAmount) return false;
+            if (token.balanceOf(this) + srcAmount > internalInventoryMax[token]) {
+                return false;
+            }
+            rateEthToToken = rateDestSrc;
+            // Normalize rate direction to enable comparison for checking spread
+            // and arbitrage: from ETH / Token -> Token / ETH
+            rateTokenToEth = 10 ** 36 / rateSrcDest;
+        }
+
+        // Check for arbitrage
+        if (rateTokenToEth < rateEthToToken) return false;
+
+        uint activationSpread = internalActivationMinSpreadBps[token];
+        return calculateSpreadBps(rateEthToToken, rateTokenToEth) >= activationSpread;
+    }
+
+    // Spread calculation is (ask - bid) / ((ask + bid) / 2).
+    // We multiply by 10000 to get result in BPS.
+    function calculateSpreadBps(
+        uint rate1,
+        uint rate2
+    )
+        public
+        pure
+        returns(uint)
+    {
+        uint diff = rate1 > rate2 ? rate1 - rate2 : rate2 - rate1;
+        return 10000 * 2 * diff / (rate1 + rate2);
```

Comment:
This calling code changed after this was implemented. New code always passes ask first.
Changed so that the function will expect ask first and return a negative int in case there is arb.